kolibri/content/utils/channel_import.py

```diff
@@ -0,0 +1,287 @@
+        return unflushed_rows
+
+    def import_channel_data(self):
+
+        unflushed_rows = 0
+
+        for model in self.content_models:
+            mapping = self.schema_mapping.get(model, {})
+            row_mapper = self.generate_row_mapper(mapping.get('per_row'))
+            table_mapper = self.generate_table_mapper(mapping.get('per_table'))
+            logging.info('Importing {model} data'.format(model=model.__name__))
+            unflushed_rows = self.table_import(model, row_mapper, table_mapper, unflushed_rows)
+        self.destination.session.commit()
+
+    def delete_content_tree_and_files(self):
+        delete_content_tree_and_files(self.channel_id)
+
+    def end(self):
+        self.source.end()
+        self.destination.end()
+
+
+class NoVersionChannelImport(ChannelImport):
+    """
+    Class defining the schema mapping for importing old content databases (i.e. ones produced before the
+    ChannelImport machinery was implemented). The schema mapping below defines how to bring in information
+    from the old version of the Kolibri content databases into the database for the current version of Kolibri.
+    """
+
+    schema_mapping = {
+        ContentNode: {
+            'per_row': {
+                'channel_id': 'infer_channel_id_from_source',
+                'tree_id': 'get_tree_id',
+                'available': 'none',
+            },
+        },
+        File: {
+            'per_row': {
+                File._meta.get_field('local_file').attname: 'checksum',
+                'available': 'none',
+            },
+        },
+        LocalFile: {
+            'per_table': 'generate_local_file_from_file',
+            'per_row': {
+                'id': 'checksum',
+                'extension': 'extension',
+                'file_size': 'file_size',
+                'available': 'none',
+            },
+        },
+        ChannelMetadata: {
+            'per_row': {
+                ChannelMetadata._meta.get_field('min_kolibri_version').attname: 'set_version_to_no_version',
+                'root_id': 'root_pk',
+            },
+        },
+    }
+
+    def none(self, source_object):
+        return None
+
+    def infer_channel_id_from_source(self, source_object):
+        return self.channel_id
+
+    def get_tree_id(self, source_object):
+        return self.tree_id
+
+    def generate_local_file_from_file(self, SourceRecord):
+        SourceRecord = self.source.get_class(File)
+        checksum_record = set()
+        # LocalFile objects are unique per checksum
+        for record in self.source.session.query(SourceRecord).all():
+            if record.checksum not in checksum_record:
+                checksum_record.add(record.checksum)
+                yield record
+            else:
+                continue
+
+    def set_version_to_no_version(self, source_object):
+        return NO_VERSION
+
+
+mappings = {
+    NO_VERSION: NoVersionChannelImport,
+    '0.6.0': ChannelImport,
```

Comment:
It currently encodes as 'version this schema was introduced in Kolibri' - only reason we would have a max version is if we had breaking changes to the schema that couldn't be filled by defaults or calculated from existing values.

Can version the schemas separately if that is more parsimonious for all.